File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceInitialize.java
package com.lumiyaviewer.lumiya.voice.common.messages;

import android.os.Bundle;
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage;

public class VoiceInitialize implements VoicePluginMessage {
    public final int appVersionCode;

    public VoiceInitialize(int appVersionCode) {
        this.appVersionCode = appVersionCode;
    }

    public VoiceInitialize(Bundle bundle) {
        this.appVersionCode = bundle.getInt("appVersionCode");
    }

    @Override
    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putInt("appVersionCode", this.appVersionCode);
        return bundle;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceInitialize.kt
package com.lumiyaviewer.lumiya.voice.common.messages

import android.os.Bundle
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage

class VoiceInitialize : VoicePluginMessage {
    @JvmField
    val appVersionCode: Int

    constructor(appVersionCode: Int) {
        this.appVersionCode = appVersionCode
    }

    constructor(bundle: Bundle) {
        this.appVersionCode = bundle.getInt("appVersionCode")
    }

    override fun toBundle(): Bundle = Bundle().apply {
        putInt("appVersionCode", appVersionCode)
    }
}
